contrib/mesos/pkg/scheduler/plugin.go

```diff
@@ -544,7 +559,8 @@ func (k *errorHandler) handleSchedulingError(pod *api.Pod, schedulingErr error)
 				defer k.api.Unlock()
 				switch task, state := k.api.tasks().Get(task.ID); state {
 				case podtask.StatePending:
-					return !task.Has(podtask.Launched) && k.api.algorithm().FitPredicate()(task, offer)
+					// TODO(sttts): why is offer matching done here outside of fcfs.go?
```

Comment:
this is an optimization: the scheduler normally "backs off" when it can't find an offer that matches up with a pod. this func assesses fitness so that the backoff period for a pod can terminate sooner if an offer becomes available that matches up.